tests/test_regression.py

```diff
@@ -0,0 +1,56 @@
+import numpy as np
+import pandas as pd
+import pytest
+
+from cleanlab.regression import rank
+
+# To be used for all the tests
+labels = np.array([1, 2, 3, 4])
+predictions = np.array([1, 3, 4, 5])
+
+# test with deafault parameters
+def test_output_shape_type():
+    scores = rank.get_label_quality_scores(labels=labels, predictions=predictions)
+    assert labels.shape == scores.shape
+    assert isinstance(scores, np.ndarray)
+
+
+# test for acceptable datatypes
+@pytest.mark.parametrize("format", [pd.Series, pd.DataFrame, list])
+def test_type_error_for_input_types(format):
+    with pytest.raises(TypeError) as error:
+        _ = rank.get_label_quality_scores(labels=format(labels), predictions=format(predictions))
+
+
+# test for input shapes
+def test_assertion_error_for_input_shape():
+    with pytest.raises(AssertionError) as error:
+        _ = rank.get_label_quality_scores(labels=labels[:-1], predictions=predictions)
+        _ = rank.get_label_quality_scores(labels=labels, predictions=predictions[:-1])
+
+
+# TODO - change name once finalised
+# test individual scoring functions
+@pytest.mark.parametrize(
+    "scoring_funcs",
+    [rank.get_residual_score_for_each_label, rank.get_score_to_named_for_each_label],
+)
+def test_individual_scoring_functions(scoring_funcs):
+    scores = scoring_funcs(labels=labels, predictions=predictions)
+    assert labels.shape == scores.shape
+    assert isinstance(scores, np.ndarray)
+
+
+# TODO - change name once finalised
+# test for method argument
+@pytest.mark.parametrize(
+    "method",
+    [
+        "residual",
+        "TO_BE_NAMED",
+    ],
+)
+def test_method_pass_get_label_quality_scores(method):
```

Comment:
This would be a perfect characterization test. As a reference, it would be nice if the scores returned by the current version of this function would not change if we were to refactor this part of the code base in the future.

You can just add an `expected_scores` as a test_parameter for each method, and finally do:

```python
assert np.allclose(scores, expected_scores, atol=1e-08)
```

Reply:
while scores with `method= "residual"` will never change for the same set of labels and predictions. But, scores with method="TO_BE_NAMED" are highly susceptible to parameters like variance, neighbors and metric for NearestNeighbors calculation. We will need to update the `expected_scores` in the test if we are changing any of these during refactoring. 

adding this as a note if we are changing any of these in the future.

Reply:
added this test in [56bc771](https://github.com/cleanlab/cleanlab/pull/572/commits/56bc77169a88041173f73dd5c15acff6a45a8417)